### ISSUE TITLE
Some chunks are not exists both on remote and local database.

### ISSUE THREAD

**Issue by scruel** [vrtmrz/obsidian-livesync#258]
## Abstract
The synchronisation hung up immediately after connecting.

## Expected behaviour
- Synchronisation ends with the message `Replication completed`
- Everything synchronised

## Actually happened
- Synchronisation has been cancelled with the message `Some chunks are not exists both on remote and local database.` (captured in the attached log, around LL.10-LL.12)
- No files synchronised

## Reproducing procedure

1. Configure LiveSync as in the attached material.
2. Create .excalidraw file and draw somthing
3. Click the replication button on the ribbon.
4. Synchronising has begun.
5. About two or three seconds later, we got the error `Some chunks are not exists both on remote and local database.`. (If not occur, repeat drawing and syncing) 
6. Replication has been stopped. No files synchronised.

### Report from the LiveSync
<details>
<summary>Report from hatch</summary>

```
----remote config----
cors:
  credentials: "true"
  headers: accept, authorization, content-type, origin, referer
  max_age: "3600"
  methods: GET,PUT,POST,HEAD,DELETE
  origins: app://obsidian.md,capacitor://localhost,http://localhost
chttpd:
  bind_address: any
  enable_cors: "true"
  max_http_request_size: "4294967296"
  port: "5984"
  require_valid_user: "true"
admins: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
vendor:
  name: The Apache Software Foundation
feature_flags:
  partitioned||*: "true"
chttpd_auth:
  authentication_redirect: /_utils/session.html
  hash_algorithms: sha256, sha
  require_valid_user: "true"
indexers:
  couch_mrview: "true"
prometheus:
  additional_port: "false"
  bind_address: 127.0.0.1
  port: "17986"
httpd:
  WWW-Authenticate: Basic realm="couchdb"
  bind_address: 0.0.0.0
  enable_cors: "true"
  port: "5986"
smoosh:
  state_dir: ./data
couch_httpd_auth:
  authentication_db: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
  secret: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
  authentication_redirect: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
couchdb_engines:
  couch: couch_bt_engine
couchdb:
  database_dir: ./data
  max_document_size: "50000000"
  single_node: "true"
  uuid: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
  view_index_dir: ./data

---- Plug-in config ---
couchDB_URI: self-hosted
couchDB_USER: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
couchDB_PASSWORD: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
couchDB_DBNAME: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
liveSync: true
syncOnSave: false
syncOnStart: false
savingDelay: 200
lessInformationInLog: false
gcDelay: 0
versionUpFlash: ""
minimumChunkSize: 20
longLineThreshold: 250
showVerboseLog: false
suspendFileWatching: false
trashInsteadDelete: true
periodicReplication: false
periodicReplicationInterval: 60
syncOnFileOpen: false
encrypt: true
passphrase: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
usePathObfuscation: false
doNotDeleteFolder: false
resolveConflictsByNewerFile: false
batchSave: false
deviceAndVaultName: ""
usePluginSettings: false
showOwnPlugins: false
showStatusOnEditor: true
usePluginSync: false
autoSweepPlugins: false
autoSweepPluginsPeriodic: false
notifyPluginOrSettingUpdated: false
checkIntegrityOnSave: false
batch_size: 50
batches_limit: 40
useHistory: true
disableRequestURI: true
skipOlderFilesOnSync: true
checkConflictOnlyOnOpen: false
syncInternalFiles: false
syncInternalFilesBeforeReplication: false
syncInternalFilesIgnorePatterns: \/node_modules\/, \/\.git\/, \/obsidian-livesync\/
syncInternalFilesInterval: 60
additionalSuffixOfDatabaseName: ""
ignoreVersionCheck: false
lastReadUpdates: 0
deleteMetadataOfDeletedFiles: false
syncIgnoreRegEx: ""
syncOnlyRegEx: \.excalidraw$|\.md$
customChunkSize: 0
readChunksOnline: true
watchInternalFileChanges: true
automaticallyDeleteMetadataOfDeletedFiles: 0
disableMarkdownAutoMerge: false
writeDocumentsIfConflicted: false
useDynamicIterationCount: false
syncAfterMerge: false
configPassphraseStore: ""
encryptedPassphrase: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
encryptedCouchDBConnection: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
permitEmptyPassphrase: false
useIndexedDBAdapter: true
useTimeouts: false
writeLogToTheFile: false
doNotPaceReplication: false
hashCacheMaxCount: 300
hashCacheMaxAmount: 50
concurrencyOfReadChunksOnline: 100
minimumIntervalOfReadChunksOnline: 333
hashAlg: xxhash64
suspendParseReplicationResult: false
doNotSuspendOnFetching: false
useIgnoreFiles: false
ignoreFiles: .gitignore

```
</details>

### Obsidian debug info
<details>
<summary>Debug info</summary>

```
Public:2023/8/13 23:05:21->Initialize done!
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:21->Lock remote database to prevent data corruption
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:22->Remote Database Destroyed
obsidian.scruel.cf/scruel-public/:1     Failed to load resource: the server responded with a status of 404 ()
plugin:obsidian-livesync:5079 The above 404 is totally normal. PouchDB is just detecting if the remote exists.
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:23->Remote Database Created or Connected
obsidian.scruel.cf/scruel-public/obsydian_livesync_version?:1     Failed to load resource: the server responded with a status of 404 ()
obsidian.scruel.cf/scruel-public/obsydian_livesync_version?:1     Failed to load resource: the server responded with a status of 404 ()
obsidian.scruel.cf/scruel-public/_local/obsydian_livesync_milestone?:1     Failed to load resource: the server responded with a status of 404 ()
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:24->Lock remote database to prevent data corruption
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:24->Hidden files and plugin synchronization have been temporarily disabled. Please enable them after the fetching, if you need them.
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:28->Hidden files and plugin synchronization have been temporarily disabled. Please enable them after the fetching, if you need them.
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:29->OneShot Sync begin... (pushOnly)
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:30->Looking for the point last synchronized point.
obsidian.scruel.cf/scruel-public/_local/WJgs0m1Qr1tBM_YPimAJKg%3D%3D?:1     Failed to load resource: the server responded with a status of 404 ()
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:30->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:30->Replication closed
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:32->OneShot Sync begin... (pushOnly)
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:32->Looking for the point last synchronized point.
obsidian.scruel.cf/scruel-public/_local/WJgs0m1Qr1tBM_YPimAJKg%3D%3D?:1     Failed to load resource: the server responded with a status of 404 ()
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:32->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:33->↑50 (34) ↓0
obsidian.scruel.cf/scruel-public/_local/WJgs0m1Qr1tBM_YPimAJKg%3D%3D?:1     Failed to load resource: the server responded with a status of 404 ()
plugin:obsidian-livesync:5079 The above 404 is totally normal. PouchDB is just checking if a remote checkpoint exists.
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:34->Synchronization setting configured as LiveSync.
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:34->Replication closed
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:34->Before LiveSync, start OneShot once...
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:34->OneShot Sync begin... (pullOnly)
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:35->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:35->Replication completed
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:35->LiveSync begin...
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:35->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->Replication closed
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->Before LiveSync, start OneShot once...
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->OneShot Sync begin... (pullOnly)
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->Replication completed
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->LiveSync begin...
2plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:36->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:47->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:05:52->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:17->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:21->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:21->DB -> STORAGE (modify,newnote) pcc_3e_slides/WORKING/python19.excalidraw
2plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:23->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:23->DB -> STORAGE (modify,newnote) pcc_3e_slides/WORKING/python19.excalidraw
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:27->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:38->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:38->Some chunks are not exists both on remote and local database.
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:38->Could not retrieve chunks
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:38->Something went wrong on reading pcc_3e_slides/WORKING/python19.excalidraw(pcc_3e_slides/WORKING/python19.excalidraw) from database:
plugin:obsidian-livesync:22510 Error: Failed to collect one of chunks
    at Object.failed (plugin:obsidian-livesync:18987:13)
    at eval (plugin:obsidian-livesync:19011:103)
    at async runWithLock (plugin:obsidian-livesync:2847:12)
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:38->Error:Failed to collect one of chunks
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:52->Replication activated
plugin:obsidian-livesync:22512 Public:2023/8/13 23:06:58->Replication is already in progress.
```
</details>

### Plug-in log

<details>
<summary>Plug-in log</summary>

```
2023/8/13 23:05:21->DB <- STORAGE (plain) pcc_3e_slides/index.md
2023/8/13 23:05:21->UPDATE DATABASE: PASS:24, FAILED:0
2023/8/13 23:05:21->Initialized, NOW TRACKING!
2023/8/13 23:05:21->Initialize done!
2023/8/13 23:05:21->Lock remote database to prevent data corruption
2023/8/13 23:05:22->Remote Database Destroyed
2023/8/13 23:05:23->Remote Database Created or Connected
2023/8/13 23:05:24->Lock remote database to prevent data corruption
2023/8/13 23:05:24->Hidden files and plugin synchronization have been temporarily disabled. Please enable them after the fetching, if you need them.
2023/8/13 23:05:28->Hidden files and plugin synchronization have been temporarily disabled. Please enable them after the fetching, if you need them.
2023/8/13 23:05:29->OneShot Sync begin... (pushOnly)
2023/8/13 23:05:30->Looking for the point last synchronized point.
2023/8/13 23:05:30->Replication activated
2023/8/13 23:05:30->Replication closed
2023/8/13 23:05:32->OneShot Sync begin... (pushOnly)
2023/8/13 23:05:32->Looking for the point last synchronized point.
2023/8/13 23:05:32->Replication activated
2023/8/13 23:05:33->↑50 (34) ↓0
2023/8/13 23:05:34->Synchronization setting configured as LiveSync.
2023/8/13 23:05:34->Replication closed
2023/8/13 23:05:34->Before LiveSync, start OneShot once...
2023/8/13 23:05:34->OneShot Sync begin... (pullOnly)
2023/8/13 23:05:35->Replication activated
2023/8/13 23:05:35->Replication completed
2023/8/13 23:05:35->LiveSync begin...
2023/8/13 23:05:35->Replication activated
2023/8/13 23:05:36->Replication closed
2023/8/13 23:05:36->Before LiveSync, start OneShot once...
2023/8/13 23:05:36->OneShot Sync begin... (pullOnly)
2023/8/13 23:05:36->Replication activated
2023/8/13 23:05:36->Replication completed
2023/8/13 23:05:36->LiveSync begin...
2023/8/13 23:05:36->Replication activated
2023/8/13 23:05:36->Replication activated
2023/8/13 23:05:47->Replication activated
2023/8/13 23:05:52->Replication activated
2023/8/13 23:06:17->Replication activated
2023/8/13 23:06:21->Replication activated
2023/8/13 23:06:21->DB -> STORAGE (modify,newnote) pcc_3e_slides/WORKING/python19.excalidraw
2023/8/13 23:06:23->Replication activated
2023/8/13 23:06:23->Replication activated
2023/8/13 23:06:23->DB -> STORAGE (modify,newnote) pcc_3e_slides/WORKING/python19.excalidraw
2023/8/13 23:06:27->Replication activated
2023/8/13 23:06:38->Replication activated
2023/8/13 23:06:38->Some chunks are not exists both on remote and local database.
2023/8/13 23:06:38->Could not retrieve chunks
2023/8/13 23:06:38->Something went wrong on reading pcc_3e_slides/WORKING/python19.excalidraw(pcc_3e_slides/WORKING/python19.excalidraw) from database:
2023/8/13 23:06:38->Error:Failed to collect one of chunks
2023/8/13 23:06:52->Replication activated
2023/8/13 23:06:58->Replication is already in progress.
```
</details>

**Comment by vrtmrz** [comment]
Thank you for opening this issue! 
This might be caused by error #257, in which we actually could not save the chunk into the database.
I am really sorry for this, but could you please let me postpone it until after #254?
